vortex-array/src/data.rs

```diff
@@ -120,7 +120,7 @@ impl From<Array> for ArrayData {
     fn from(value: Array) -> ArrayData {
         match &value {
             Array::Data(d) => d.clone(),
-            Array::View(_) => value.clone().into(),
```

Comment:
The broken thing is here. This is an infinite recursion.

Reply:
I don't mind it here, mostly concerned about the amount of small traits we have, which makes it much harder to discover functionality like this. Can't we just do an explicit `from` instead?

Reply:
There is no from function to call. I tried making it work but I think we do need a trait. Could be convinced otherwise but I couldn’t figure out how to make it work